src/main/java/org/scijava/ops/matcher/DefaultOpTypeMatchingService.java

```diff
@@ -239,15 +241,9 @@ private boolean typesPerfectMatch(final OpCandidate candidate) {
 			i++;
 		}
 
-		Type[] outputTypes = candidate.getRef().getOutTypes();
-		i = 0;
-		for (final Type t : OpUtils.outputTypes(candidate)) {
-			if (outputTypes[i] != null) {
-				if (!t.equals(outputTypes[i]))
-					return false;
-			}
-			i++;
-		}
+		final Type outputType = candidate.getRef().getOutType();
+		if (Objects.equal(outputType, OpUtils.outputType(candidate)))
```

Comment:
Do you need the Guava `Objects.equal` here, rather than the standard Java library's `Objects`?

Reply:
Woops, nope, that probably was an auto-import fail. Good catch - thanks!